### DISCUSSION TITLE
infinity loop to the use setError with onChangeText

### DISCUSSION THREAD

**Discussion by julio3266** [react-hook-form/react-hook-form#7812] [Show and tell] [answered]
I'm having a one problem to the use setError with onChangeText, if i use onChange everything happens normal
OBS: the validation is going correctly
React Hook Form Version: 6.9.22 

my input with controller:
![code](https://user-images.githubusercontent.com/52897217/153767403-5da406e2-ec2b-4e9d-852d-f936e9aa086b.png)

**Comment by Moshyfawn** [comment, marked as answer]
We appreciate you taking the time to screenshot your code, but it doesn't exactly help to debug the issue like [a codesanbox](https://codesandbox.io/s/react-hook-form-v7-ts-template-4tn8i) with an isolated reproduction does.

**Reply by Moshyfawn** [reply]
As a side note, it's recommended to use [the `rules` prop](https://react-hook-form.com/v6/api#Controller) with the validation fields or a custom `validate` function instead of `setError` for such use
```js
rules={{
  required: "firstName is required",
  // OR
  validate: (value) => {
    if (!value) {
      return "firstName is required";
    }
    return true;
  }
}}
```

**Reply by julio3266** [reply]
@Moshyfawn i resolved my problem, just passing ref={register(...)}, but I have a one question... can I pass the type of validation for one input? 
I have a input for  birth of date
I solved my problem, just passing ref={register(...)}, but I have a question... can I pass the validation type to an input?
I have an entry for date of birth that does not use onChange but onFocus, it is a modal, this validation is not appearing for me
![code](https://user-images.githubusercontent.com/52897217/153771301-88103102-999f-4629-9cdd-0a26fdea6ac3.png)

**Reply by Moshyfawn** [reply]
You shouldn't use `register` for **controlled** components that use `Controller`. Some unwanted behavior is bound to happen.
To apply validation to `Controller`, use [the `rules` prop](https://react-hook-form.com/v6/api#Controller) and pass the validation you've applied via `register` into the `Controller`'s rules instead.

In regards to validation `onFocus`, RHF doesn't support such behavior. More info on the validation modes you can [see here](https://react-hook-form.com/api/useform#props)
If I understand you correctly _(and again, I can't be sure without seeing an actual use-case reproduction)_, you can pass [the trigger form method](https://react-hook-form.com/api/useform/trigger) into your `onFocus` and call it with the name of the input you want to validate.
```js
onFocus(() => {
  trigger("fieldName")
})
```

**Reply by julio3266** [reply]
@Moshyfawn thx you for your help, this solution resolved my problem <3

**Reply by bluebill1049** [reply]
thank you @Moshyfawn 🙏
